lldb/test/API/functionalities/multiple-slides/TestMultipleSlides.py

```diff
@@ -12,6 +12,7 @@
 class MultipleSlidesTestCase(TestBase):
     NO_DEBUG_INFO_TESTCASE = True
 
+    @expectedFailureAll(oslist=["windows"], archs=["x86_64"])
```

Comment:
Could you provide a failure log for this? It passes on our CI (Windows native x86_64).

```
********************
PASS: lldb-api :: functionalities/multiple-slides/TestMultipleSlides.py (530 of 1191)
Script:
--
C:/Python312/python.exe C:/lldb\test\API\dotest.py -u CXXFLAGS -u CFLAGS --env OBJCOPY=C:/build-lldb-native/./bin/llvm-objcopy.exe --env STRIP=C:/build-lldb-native/./bin/llvm-strip.exe --env LLVM_LIBS_DIR=C:/build-lldb-native/./lib --env LLVM_INCLUDE_DIR=C:/build-lldb-native/include --env LLVM_TOOLS_DIR=C:/build-lldb-native/./bin --arch x86_64 --build-dir C:/build-lldb-native/lldb-test-build.noindex --lldb-module-cache-dir C:/build-lldb-native/lldb-test-build.noindex/module-cache-lldb\lldb-api --clang-module-cache-dir C:/build-lldb-native/lldb-test-build.noindex/module-cache-clang\lldb-api --executable C:/build-lldb-native/./bin/lldb.exe --make c:/gl/b1/3rdpaty/bin/make-wfix.exe --compiler C:/build-lldb-native/./bin/clang.exe --dsymutil C:/build-lldb-native/./bin/dsymutil.exe --llvm-tools-dir C:/build-lldb-native/./bin --lldb-obj-root C:/build-lldb-native/tools/lldb --lldb-libs-dir C:/build-lldb-native/./lib --skip-category=watchpoint C:\lldb\test\API\functionalities\multiple-slides -p TestMultipleSlides.py
--
Exit Code: 0

Command Output (stdout):
--
lldb version 20.0.0git (https://gitlab-ci-token:glcbt-64_rzNTyT5_UDR5sLkLB8b6@gitlab.ninolab.accesssoftek.com/accesssoftek/lldb-test-scripts.git revision cbd39658b6aed4dbb66a2b775096974c9087e93d)
  clang revision cbd39658b6aed4dbb66a2b775096974c9087e93d
  llvm revision cbd39658b6aed4dbb66a2b775096974c9087e93d
Skipping the following test categories: ['watchpoint', 'libc++', 'libstdcxx', 'dwo', 'dsym', 'gmodules', 'debugserver', 'objc', 'fork', 'pexpect']


--
Command Output (stderr):
--
PASS: LLDB (C:\build-lldb-native\bin\clang.exe-x86_64) :: test_mulitple_slides (TestMultipleSlides.MultipleSlidesTestCase.test_mulitple_slides)

----------------------------------------------------------------------

Ran 1 test in 2.447s



OK


--

```

Reply:
Also, it may be worth cherry-picking this https://github.com/llvm/llvm-project/commit/58d22f1b665ca8810c4de341426247b7b5a69548 to have build commands and outputs in the log without tracing hassle.

Reply:
Ty! I'll try that out.

Reply:
Looks like after my CMake chages,this one is passing now.  I'll remove it from the PR

Reply:
Ah, sorry I spoke too soon. On windows I am seeing this fail. The test is built without debug information because this is in it's makefile:

```
C_SOURCES := main.c
MAKE_DSYM := NO

include Makefile.rules

main.o: main.c
	$(CC) $(CFLAGS_NO_DEBUG) -c $< -o $@
```

I am curious to know how this is passing for you? Without specifying `-g` or `-gdwarf` to Clang, lldb cannot resolve the name `first` and fails for me locally. But adding either of those flags fixes the problem.

Reply:
Here's the execution log on my setup https://gist.github.com/dzhidzhoev/c6af2c83c6175eb1ba83c306b0ad9a15.

Build commands are
```
F:/Users/vdzhidzhoev/temp/build-lldb-native/bin/clang.exe -O0 -m64     -c F:\Users\vdzhidzhoev\temp\llvm-project\lldb\test\API\functionalities\multiple-slides/main.c -o main.o
F:/Users/vdzhidzhoev/temp/build-lldb-native/bin/clang.exe main.o -gdwarf -O0  -m64 -IF:\Users\vdzhidzhoev\temp\llvm-project\lldb\packages\Python\lldbsuite\test\make/../../../../..//include -IF:/Users/vdzhidzhoev/temp/build-lldb-native/tools/lldb/include -IF:\Users\vdzhidzhoev\temp\llvm-project\lldb\test\API\functionalities\multiple-slides -IF:\Users\vdzhidzhoev\temp\llvm-project\lldb\packages\Python\lldbsuite\test\make -include F:\Users\vdzhidzhoev\temp\llvm-project\lldb\packages\Python\lldbsuite\test\make/test_common.h -fno-limit-debug-info     -fuse-ld=lld  --driver-mode=g++ -o "a.out"
```

It seems that -gdwarf is added on this line https://github.com/llvm/llvm-project/blob/8f7910a4fc08818462e0e836db74b20290cd36c7/lldb/packages/Python/lldbsuite/test/make/Makefile.rules#L270. 

Here's objdump for main.o:
```
$ bin/llvm-objdump -t  /f/Users/vdzhidzhoev/temp/build-lldb-native/lldb-test-build.noindex/functionalities/multiple-slides/TestMultipleSlides.test_mulitple_slides/main.o  | less

F:/Users/vdzhidzhoev/temp/build-lldb-native/lldb-test-build.noindex/functionalities/multiple-slides/TestMultipleSlides.test_mulitple_slides/main.o:     file format coff-x86-64

SYMBOL TABLE:
[ 0](sec  1)(fl 0x00)(ty   0)(scl   3) (nx 1) 0x00000000 .text
AUX scnlen 0x17 nreloc 2 nlnno 0 checksum 0xdb640681 assoc 1 comdat 0
[ 2](sec  2)(fl 0x00)(ty   0)(scl   3) (nx 1) 0x00000000 .data
AUX scnlen 0x4000 nreloc 0 nlnno 0 checksum 0x8ff2bf4e assoc 2 comdat 0
[ 4](sec  3)(fl 0x00)(ty   0)(scl   3) (nx 1) 0x00000000 .bss
AUX scnlen 0x0 nreloc 0 nlnno 0 checksum 0x0 assoc 3 comdat 0
[ 6](sec  4)(fl 0x00)(ty   0)(scl   3) (nx 1) 0x00000000 .xdata
AUX scnlen 0x8 nreloc 0 nlnno 0 checksum 0x1ab96b84 assoc 4 comdat 0
[ 8](sec  5)(fl 0x00)(ty   0)(scl   3) (nx 1) 0x00000000 .pdata
AUX scnlen 0xc nreloc 3 nlnno 0 checksum 0x767e3832 assoc 5 comdat 0
[10](sec  6)(fl 0x00)(ty   0)(scl   3) (nx 1) 0x00000000 .llvm_addrsig
AUX scnlen 0x2 nreloc 0 nlnno 0 checksum 0xe3c301f assoc 6 comdat 0
[12](sec -1)(fl 0x00)(ty   0)(scl   3) (nx 0) 0x00000000 @feat.00
[13](sec  1)(fl 0x00)(ty  20)(scl   2) (nx 0) 0x00000000 main
[14](sec  2)(fl 0x00)(ty   0)(scl   2) (nx 0) 0x00000000 first
[15](sec  2)(fl 0x00)(ty   0)(scl   2) (nx 0) 0x00002000 second
[16](sec -2)(fl 0x00)(ty   0)(scl  67) (nx 1) 0x00000000 .file

```

Here's "image dump symtab" from lldb.exe for a.out https://gist.github.com/dzhidzhoev/ee3bd2566f3a95b018f40c45e34d9078.

Reply:
"first" is preset in both dumps

Reply:
I see. Our command lines look identical. Can you also please share the linker command that your clang invocation is generating? I am also curious to know how you built lld. Perhaps I'm missing an earlier step.

My clang output with `-v` specified is:

```
"C:\workspace\llvm-project\build\bin\clang.exe" -O0 -m64    -c C:\workspace\llvm-project\lldb\test\API\functionalities\multiple-slides/main.c -o main.o
"C:\workspace\llvm-project\build\bin\clang.exe" main.o -gdwarf -O0 -m64 -IC:\workspace\llvm-project\lldb\packages\Python\lldbsuite\test\make/../../../../../include -IC:/workspace/llvm-project/build/tools/lldb/include -IC:\workspace\llvm-project\lldb\test\API\functionalities\multiple-slides -IC:\workspace\llvm-project\lldb\packages\Python\lldbsuite\test\make -include C:\workspace\llvm-project\lldb\packages\Python\lldbsuite\test\make/test_common.h -fno-limit-debug-info   -v  -fuse-ld=lld --driver-mode=g++ -o "a.out"
clang version 20.0.0git (https://github.com/kendalharland/llvm-project bf6f1f2f5f23857eb18e08fbef29c5ca2e553bea)
Target: x86_64-unknown-windows-msvc
Thread model: posix
InstalledDir: C:\workspace\llvm-project\build\bin
 "C:\\workspace\\llvm-project\\build\\bin\\lld-link" -out:a.out -defaultlib:libcmt -defaultlib:oldnames -nologo -debug main.o
```

Reply:
For comparison, my objdump output shows no symbol table:

```
PS C:\workspace\llvm-project> C:\workspace\llvm-project\build\bin\llvm-objdump -t C:\workspace\llvm-project\build\lldb-test-build.noindex\functionalities\multiple-slides\TestMultipleSlides.test_mulitple_slides\a.out

C:\workspace\llvm-project\build\lldb-test-build.noindex\functionalities\multiple-slides\TestMultipleSlides.test_mulitple_slides\a.out:  file format coff-x86-64

SYMBOL TABLE:
```

Reply:
```
$ F:/Users/vdzhidzhoev/temp/build-lldb-native/bin/clang.exe main.o -gdwarf -O0  -m64 -IF:\Users\vdzhidzhoev\temp\llvm-project\lldb\packages\Python\lldbsuite\test\make/../../../../..//include -IF:/Users/vdzhidzhoev/temp/build-lldb-native/tools/lldb/include -IF:\Users\vdzhidzhoev\temp\llvm-project\lldb\test\API\functionalities\multiple-slides -IF:\Users\vdzhidzhoev\temp\llvm-project\lldb\packages\Python\lldbsuite\test\make -include F:\Users\vdzhidzhoev\temp\llvm-project\lldb\packages\Python\lldbsuite\test\make/test_common.h -fno-limit-debug-info     -fuse-ld=lld  --driver-mode=g++ -o "a.out" -v

clang version 20.0.0git (git@gitlab.ninolab.accesssoftek.com:accesssoftek/lldb-test-scripts.git 8d8d23fec72ad6ff77e15e7627ab8cc26ac36d4c)
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: F:\Users\vdzhidzhoev\temp\build-lldb-native\bin
Build config: +assertions
 "F:\\Users\\vdzhidzhoev\\temp\\build-lldb-native\\bin\\lld-link" -out:a.out -defaultlib:libcmt -defaultlib:oldnames -nologo -debug main.o
```

Reply:
> For comparison, my objdump output shows no symbol table:
> 
> ```
> PS C:\workspace\llvm-project> C:\workspace\llvm-project\build\bin\llvm-objdump -t C:\workspace\llvm-project\build\lldb-test-build.noindex\functionalities\multiple-slides\TestMultipleSlides.test_mulitple_slides\a.out
> 
> C:\workspace\llvm-project\build\lldb-test-build.noindex\functionalities\multiple-slides\TestMultipleSlides.test_mulitple_slides\a.out:  file format coff-x86-64
> 
> SYMBOL TABLE:
> ```

I have the same for a.out. The dump I provided (with "first") was launched for main.o.
Try launching bin/lldb, "file a.out", "image dump symtab a.out", and there should be "first" symbol. Beware that will probably show A LOT of entries.

Reply:
My output for `image dump symtab a.out` is empty. However, if I edit https://github.com/llvm/llvm-project/blob/b66aa3bfff442a5eb67f1bfcfaa148e42e49b787/lldb/test/API/functionalities/multiple-slides/Makefile#L11 to use CFLAGS instead of CFLAGS_NO_DEBUG lldb I can see symtab output similar to yours above. Do you have happen to be working with a patched or older version of that file, or have settings in your environment that force `-gdwarf` to be passed to clang? I ask because although I am now able to see debug info, I still am wondering if there's something in my local environment that is causing different behavior. Additionally, LLDB produces no output `image lookup -s first`, which still breaks the test locally but it can resolve "first" if I instead run `ta v first`. Something is preventing it from looking up the symbols for `first` and `second` even though they are clearly present as globals

Reply:
I was able to figure this out. My CMakeCache had `-D LLVM_ENABLE_DIA_SDK="OFF"` which I must have set in a previous build. I don't recall why I set this, must have been trying to repro someone else's build and forgot. This indeed passes for me now. Thanks so much for your patience and helping me debug!